runtime/io.c

```diff
@@ -203,6 +217,8 @@ CAMLexport void caml_putword(struct channel *channel, uint32_t w)
 CAMLexport int caml_putblock(struct channel *channel, char *p, intnat len)
 {
   int n, free, towrite, written;
+ again:
+  check_pending(channel);
```

Comment:
I think there is a problem with this loop and the interaction with the GC. Indeed, the parameter `p` is called with `&Byte(buff, pos)`, which means that it points to something in the heap which may move when signal handler are executed.

(Note that the loop existed in `unix.c` before that PR, but this is still a bug, so let's fix it now.)

Reply:
I think this was already fixed by f43daf93da37f9f1a691a263d7173156fdd445ea ?

Reply:
Indeed.